// Generated by ReScript, PLEASE EDIT WITH CARE
'use strict';

var Caml_exceptions = require("rescript/lib/js/caml_exceptions.js");

var p1 = new Promise((function (resolve, _reject) {
        return resolve("hello world");
      }));

var p2 = Promise.resolve("some value");

var MyOwnError = /* @__PURE__ */Caml_exceptions.create("AsyncAndPromise.MyOwnError");

var p3 = Promise.reject({
      RE_EXN_ID: MyOwnError,
      _1: "some rejection"
    });

Promise.resolve("hello world").then(function (msg) {
        return Promise.resolve("Message: " + msg);
      }).then(function (msg) {
      console.log(msg);
      return Promise.resolve(undefined);
    });

exports.p1 = p1;
exports.p2 = p2;
exports.MyOwnError = MyOwnError;
exports.p3 = p3;
/* p1 Not a pure module */
